Assignment4/CarlosD21-190382/Task07.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1tV5j-DRcpPtoJGoMj8v2DSqR_9wyXeiE
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+print("RDFLib:")
+#Subclass
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s,p,o)
+
+print("SPARQL:")
+q1 = prepareQuery(''' SELECT ?sub WHERE { 
+    ?sub rdfs:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns": ns}
+)
+
+for r in g.query(q1):
+  print(r)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**"""
+
+from rdflib.plugins.sparql import prepareQuery
+print("RDFLib:")
+ns = Namespace("http://somewhere#")
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
+#Subclass
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s2,p2,o2 in g.triples((None, RDF.type, s)):
+    print(s2)
+
+print("SPARQL:")
+q2= prepareQuery('''SELECT ?sub WHERE { 
+    { 
+      ?sub rdf:type ns:Person. 
+    } 
+    UNION {
+      ?subclass rdfs:subClassOf ns:Person.
+      ?sub rdf:type ?subclass
```

Comment:
Missing all the subclass hierarchy (use chained properties)

Reply:
I have done some changes but I don´t understand very well this point.